sdk/go/common/resource/plugin/plugin.go

```diff
@@ -372,13 +372,22 @@ func execPlugin(ctx *Context, bin, prefix string, kind apitype.PluginKind,
 
 		logging.V(9).Infof("Launching plugin '%v' from '%v' via runtime '%s'", prefix, pluginDir, runtimeInfo.Name())
 
-		// ProgramInfo needs pluginDir to be an absolute path
-		pluginDir, err = filepath.Abs(pluginDir)
+		// Ensure the plugin directory is an absolute path.
+		pluginDir, err := filepath.Abs(pluginDir)
 		if err != nil {
 			return nil, fmt.Errorf("getting absolute path for plugin directory: %w", err)
 		}
+		if len(env) == 0 {
+			// If no environment is provided, use the current environment.
+			env = os.Environ()
+		}
+		info := NewProgramInfo(rootDir, pluginDir, ".", runtimeInfo.Options())
```

Comment:
This should be the "plugins" root dir, which currently always == pluginDir

Reply:
It is. We pass in the same thing for rootDir and pluginDir currently. The reason for passing rootDir here is just for future proofing.

Reply:
Yeh I don't think you need that future proofing, and this is all internal code anyway so we can fix it up later if it comes up.